src/accounting.rs

```diff
@@ -2,119 +2,168 @@ use cketh_common::eth_rpc_client::providers::RpcApi;
+    let nodes_in_subnet = UNSTABLE_SUBNET_SIZE.with(|n| *n.borrow());
+    let ingress_bytes = (payload_size_bytes + api.url.len()) as u128 + INGRESS_OVERHEAD_BYTES;
     let base_cost = INGRESS_MESSAGE_RECEIVED_COST
         + INGRESS_MESSAGE_BYTE_RECEIVED_COST * ingress_bytes
         + HTTP_OUTCALL_REQUEST_COST
         + HTTP_OUTCALL_BYTE_RECEIEVED_COST * (ingress_bytes + max_response_bytes as u128);
-    base_cost * (nodes_in_subnet as u128) / DEFAULT_NODES_IN_SUBNET as u128
+    base_cost * (nodes_in_subnet as u128) / NODES_IN_DEFAULT_SUBNET as u128
 }
 
 /// Calculate the additional cost for calling a registered JSON-RPC provider.
-pub fn get_provider_cost(provider: &Provider, json_rpc_payload: &str) -> u128 {
+pub fn get_provider_cost(provider: &Provider, payload_size_bytes: usize) -> u128 {
     let nodes_in_subnet = UNSTABLE_SUBNET_SIZE.with(|m| *m.borrow());
     let cost_per_node = provider.cycles_per_call as u128
-        + provider.cycles_per_message_byte as u128 * json_rpc_payload.len() as u128;
+        + provider.cycles_per_message_byte as u128 * payload_size_bytes as u128;
     cost_per_node * (nodes_in_subnet as u128)
 }
 
 #[test]
 fn test_request_cost() {
-    let url = "https://cloudflare-eth.com";
-    let payload = "{\"jsonrpc\":\"2.0\",\"method\":\"eth_gasPrice\",\"params\":[],\"id\":1}";
-    let base_cost = get_request_cost(
-        &ResolvedJsonRpcSource::Api(RpcApi {
-            url: url.to_string(),
-            headers: vec![],
-        }),
-        payload,
-        1000,
-    );
-    let s10 = "0123456789";
-    let base_cost_s10 = get_request_cost(
-        &ResolvedJsonRpcSource::Api(RpcApi {
-            url: url.to_string(),
-            headers: vec![],
-        }),
-        &(payload.to_string() + s10),
-        1000,
-    );
-    assert_eq!(
-        base_cost + 10 * (INGRESS_MESSAGE_BYTE_RECEIVED_COST + HTTP_OUTCALL_BYTE_RECEIEVED_COST),
-        base_cost_s10
-    )
+    for nodes_in_subnet in [1, NODES_IN_DEFAULT_SUBNET, NODES_IN_FIDUCIARY_SUBNET] {
+        println!("Nodes in subnet: {nodes_in_subnet}");
+
+        UNSTABLE_SUBNET_SIZE.with(|n| *n.borrow_mut() = nodes_in_subnet);
+
+        let url = "https://cloudflare-eth.com";
+        let payload = "{\"jsonrpc\":\"2.0\",\"method\":\"eth_gasPrice\",\"params\":[],\"id\":1}";
+        let base_cost = get_json_rpc_cost(
+            &ResolvedJsonRpcSource::Api(RpcApi {
+                url: url.to_string(),
+                headers: vec![],
+            }),
+            payload.len(),
+            1000,
+        );
+        let base_cost_10_extra_bytes = get_json_rpc_cost(
+            &ResolvedJsonRpcSource::Api(RpcApi {
+                url: url.to_string(),
+                headers: vec![],
+            }),
+            payload.len() + 10,
+            1000,
+        );
+        let estimated_cost_10_extra_bytes = base_cost
+            + 10 * (INGRESS_MESSAGE_BYTE_RECEIVED_COST + HTTP_OUTCALL_BYTE_RECEIEVED_COST)
```

Comment:
`HTTP_OUTCALL_BYTE_RECEIEVED_COST` -> `HTTP_OUTCALL_BYTE_RECEIVED_COST`

Reply:
Thanks for catching that. For the record, this typo was in the original canister before I started working on it (along with many others).